File: frontend/src/services/supabase.js
import { createClient } from '@supabase/supabase-js'

// Načtení proměnných prostředí z .env souboru
const supabaseUrl = import.meta.env.VITE_SUPABASE_URL || 'http://localhost:8000'
const supabaseKey = import.meta.env.VITE_SUPABASE_PUBLIC_KEY

// Kontrola, zda jsou nastavené proměnné prostředí
if (!supabaseKey) {
  console.warn('VITE_SUPABASE_PUBLIC_KEY není nastaven. Použijte .env soubor pro nastavení přístupových údajů.')
}

// Vytvoření Supabase klienta
const supabase = createClient(supabaseUrl, supabaseKey || 'public-anon-key')

// Funkce pro testování připojení k Supabase
export const testConnection = async () => {
  try {
    const { data, error } = await supabase.from('users').select('count').limit(1)
    if (error) throw error
    console.log('Supabase připojení úspěšné:', data)
    return true
  } catch (error) {
    console.error('Chyba při připojování k Supabase:', error.message)
    return false
  }
}

export default supabase

File: backend/app/core/supabase.py
from supabase import create_client, Client
from .config import settings
import logging

# Nastavení loggeru
logger = logging.getLogger(__name__)

# Vytvoření Supabase klienta
try:
    supabase: Client = create_client(
        settings.SUPABASE_URL,
        settings.SUPABASE_KEY
    )
    logger.info(f"Supabase klient vytvořen pro URL: {settings.SUPABASE_URL}")
except Exception as e:
    logger.error(f"Chyba při vytváření Supabase klienta: {str(e)}")
    # Vytvoření prázdného klienta pro případ chyby
    supabase = None

def get_supabase_client() -> Client:
    """
    Vrátí instanci Supabase klienta pro použití v API endpointech.

    Returns:
        Client: Supabase klient nebo None v případě chyby

    Raises:
        Exception: Pokud není Supabase klient inicializován
    """
    if supabase is None:
        raise Exception("Supabase klient není inicializován. Zkontrolujte přístupové údaje v .env souboru.")
    return supabase

# Funkce pro testování připojení k Supabase
async def test_connection() -> bool:
    """
    Otestuje připojení k Supabase.

    Returns:
        bool: True pokud je připojení úspěšné, jinak False
    """
    try:
        if supabase is None:
            return False

        # Zkusíme získat verzi PostgreSQL
        response = await supabase.rpc('get_postgres_version').execute()
        if response.error:
            logger.error(f"Chyba při testování připojení k Supabase: {response.error.message}")
            return False

        logger.info(f"Připojení k Supabase úspěšné: {response.data}")
        return True
    except Exception as e:
        logger.error(f"Chyba při testování připojení k Supabase: {str(e)}")
        return False
